toml/src/main/kotlin/org/rust/toml/completion/CargoTomlCompletionContributor.kt

```diff
@@ -7,25 +7,38 @@ package org.rust.toml.completion
 
 import com.intellij.codeInsight.completion.CompletionContributor
 import com.intellij.codeInsight.completion.CompletionType.BASIC
+import org.rust.ide.experiments.RsExperiments
+import org.rust.openapiext.isFeatureEnabled
 import org.rust.toml.CargoTomlPsiPattern.inDependencyKeyValue
 import org.rust.toml.CargoTomlPsiPattern.inDependencyPackageFeatureArray
 import org.rust.toml.CargoTomlPsiPattern.inFeatureDependencyArray
 import org.rust.toml.CargoTomlPsiPattern.inKey
 import org.rust.toml.CargoTomlPsiPattern.inSpecificDependencyHeaderKey
 import org.rust.toml.CargoTomlPsiPattern.inSpecificDependencyKeyValue
 import org.rust.toml.CargoTomlPsiPattern.inValueWithKey
+import org.rust.toml.crates.local.completion.DependencyCratesLocalCompletionProvider
 import org.rust.toml.tomlPluginIsAbiCompatible
 
 class CargoTomlCompletionContributor : CompletionContributor() {
     init {
         if (tomlPluginIsAbiCompatible()) {
             extend(BASIC, inKey, CargoTomlKeysCompletionProvider())
             extend(BASIC, inValueWithKey("edition"), CargoTomlKnownValuesCompletionProvider(listOf("2015", "2018")))
-            extend(BASIC, inDependencyKeyValue, CargoTomlDependencyCompletionProvider())
             extend(BASIC, inSpecificDependencyHeaderKey, CargoTomlSpecificDependencyHeaderCompletionProvider())
             extend(BASIC, inSpecificDependencyKeyValue, CargoTomlSpecificDependencyVersionCompletionProvider())
             extend(BASIC, inFeatureDependencyArray, CargoTomlFeatureDependencyCompletionProvider())
             extend(BASIC, inDependencyPackageFeatureArray, CargoTomlDependencyFeaturesCompletionProvider())
+
+            val isCratesLocalIndexEnabled = isFeatureEnabled(RsExperiments.CRATES_LOCAL_INDEX)
+
+            extend(
+                BASIC, inDependencyKeyValue,
+
+                if (isCratesLocalIndexEnabled)
+                    DependencyCratesLocalCompletionProvider()
+                else
+                    CargoTomlDependencyCompletionProvider()
+            )
```

Comment:
IMO something like that looks a bit clearer
```Kotlin
val dependencyCompletionProvider = if (isFeatureEnabled(RsExperiments.CRATES_LOCAL_INDEX)) {
    DependencyCratesLocalCompletionProvider()
} else {
    CargoTomlDependencyCompletionProvider()
}
extend(BASIC, inKey, CargoTomlKeysCompletionProvider())
// ...
extend(BASIC, inDependencyKeyValue, dependencyCompletionProvider)
```

Reply:
I was thinking what if we would have several of them in the future. Then we will have to add a variable with the condition for each of the completion we implement both for API and Crates Local Index until e.g. crate resolver is implemented